### DISCUSSION TITLE
Syntax Highlighting For *.fpp Files

### DISCUSSION THREAD

**Discussion by dmitri-mcguckin** [nasa/fprime#1849] [Ideas]
Hello, I just wanted to throw this idea out there, as I've not seen any topic related to this in forums or issues.

But if there is a pre-existing thread on it, please do point me to it and I can close this thread.

I was wondering if there have been any efforts to create syntax highlighting options/plugins for popular IDEs such as VSCode or Atom.

If not, I wanted to see in what ways I can contribute that, myself.

I'd certainly be willing to collaborate on such a thing, too, since I'm by no means an expert on the grammars that build the FPP language.

So let me know your thoughts,
All the best.

**Comment by astroesteban** [comment]
Hey @dmitri-mcguckin 👋🏼 

 I've been using the following FPP extension for VS Code:
https://marketplace.visualstudio.com/items?itemName=unlv-team5.fpptools

And the repo is here:
https://github.com/phxe/fpptools

I'm sure the maintainers would also appreciate any contributions to the extension.

**Reply by dmitri-mcguckin** [reply]
Hey, thanks for pointing that out!

The default instructions listed on the marketplace don't seem to work as it doesn't seem to show up. It may be unregistered? Not sure.

But incase this is useful for anyone else that finds this:
* Went to the [FPP Tools Releases](https://github.com/phxe/fpptools/releases)
* Downloaded the latest release of the `*.vsix` file
* Used the manual "Install from vsix" option in vscode

![vscode](https://user-images.githubusercontent.com/28746912/212765468-fe5577e5-9820-438c-9db1-db66eaa3ede8.jpeg)

The highlighting seems to work like a charm as is.

**Reply by astroesteban** [reply]
That's odd. I had just installed it last week from the extensions pane in VS Code. Not sure why it didn't work for you but the instructions you provided are useful 👍🏼

**Reply by ThibFrgsGmz** [reply]
Personally, I can't always find the FPPTools in the VSCode search. I use most of the time the installation by VSIX. It is not necessarily annoying but the extension is not known enough.

**Reply by astroesteban** [reply]
Weird. That might be a bug report we'd have to file with the project maintainers. Although I don't know how active it is. Last update was 8 months ago...
